### PR TITLE
BRD - ABC override to ignore army's paeon + muse

### PR DESCRIPTION
**Pull request by yumiyafangirl**
All GCDs under one of bard's RNG speed mods are dropped from the uptime calculation altogether.

Before:
image

After:
image

### CODE REVIEW

File: src/parser/jobs/brd/modules/AlwaysBeCasting.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	protected getUptimePercent(): number {
+		const fightDuration = this.parser.currentDuration - this.downtime.getDowntime()
+		const armyDuration = this.armyHistory.reduce((acc, army) => {
+			const downtime = this.downtime.getDowntime(army.start, army.end)
+			return acc + army.end - army.start - downtime
+		}, 0)
+
+		const uptime = this.gcd.gcds.reduce((acc, gcd) => {
+			const duration = this.gcd._getGcdLength(gcd)
```
Comment (ayyaruq) [inline src/parser/jobs/brd/modules/AlwaysBeCasting.ts:77]: I think this is all kinds of inaccurate but yolo

File: src/parser/jobs/brd/modules/index.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import AdditionalStats from './AdditionalStats'
 import ApexArrow from './ApexArrow'
+import BardABC from './AlwaysBeCasting'
```
Comment (ayyaruq) [inline src/parser/jobs/brd/modules/index.js:3]: Bruh

File: src/parser/jobs/brd/modules/index.js [hunk 3]
```diff
@@ -15,6 +16,7 @@ import Weaving from './Weaving'
 export default [
 	AdditionalStats,
 	ApexArrow,
+	BardABC,
```
Comment (ayyaruq) [inline src/parser/jobs/brd/modules/index.js:19]: Same

**Review by ayyaruq — CHANGES_REQUESTED** [review]
Code-wise it's fine even if I think the premise is silly but at least it'll be more consistent. Need you to fix those imports tho.

File: src/parser/jobs/brd/modules/AlwaysBeCasting.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import AlwaysBeCasting from 'parser/core/modules/AlwaysBeCasting'
```
[the remark this answers is not in the document]
Comment (yumiyafangirl) [reply src/parser/jobs/brd/modules/AlwaysBeCasting.ts]: Good tip, I corrected this and also opened https://github.com/xivanalysis/xivanalysis/pull/1039 to fix another instance of bad import naming

File: src/parser/jobs/brd/modules/AlwaysBeCasting.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	private onSong(event: CastEvent) {
+		if (event.ability.guid === ACTIONS.ARMYS_PAEON.id) {
+			this.currentPaeon = {
+				start: event.timestamp,
+				end: Math.min(this.parser.fight.end_time, event.timestamp + SONG_DURATION_MS),
+			}
+		} else if (this.currentPaeon) {
+			this.currentPaeon.end = event.timestamp
+			this.endPaeon()
+		}
```
[the remark this answers is not in the document]
Comment (yumiyafangirl) [reply src/parser/jobs/brd/modules/AlwaysBeCasting.ts:63]: The 'removebuff' hook catches the case where they let the song expire naturally, so I think we're fine here.

File: src/parser/jobs/brd/modules/AlwaysBeCasting.ts (hunk 1 above)
Comment (yumiyafangirl) [reply src/parser/jobs/brd/modules/AlwaysBeCasting.ts:78]: ye

File: src/parser/jobs/brd/modules/AlwaysBeCasting.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ayyaruq) [reply src/parser/jobs/brd/modules/AlwaysBeCasting.ts:63]: Good enough I guess, yolo.

**Review by ayyaruq — DISMISSED** [review]
Noice, well actually kinda gross but it gets the job done. WTB accurate GCD details.

**Review by ayyaruq — APPROVED** [review]
Boom, we did it reddit

**Pull request merged**